lib/queuePopulator/IngestionProducer.js

```diff
@@ -41,26 +45,53 @@ class IngestionProducer {
     constructor(sourceConfig, qpConfig, s3Config) {
         this.log = new Logger('Backbeat:IngestionProducer');
         this.qpConfig = qpConfig;
-        this.s3sourceCredentials = new AWS.Credentials({
-            accessKeyId: sourceConfig.auth.accessKey,
-            secretAccessKey: sourceConfig.auth.secretKey,
-        });
-
-        const { https, host, port } = sourceConfig;
-        const protocol = https ? 'https' : 'http';
-
         this.s3source = s3Config;
         this._targetZenkoBucket = sourceConfig.name;
         this.requestLogger = this.log.newRequestLogger();
         this.createEntry = new RaftLogEntry();
-        this.resLog = [];
-        this.sourceHTTPAgent = new http.Agent({ keepAlive: true });
 
-        this.ringReader = new BackbeatClient({
-            endpoint: `${protocol}://${host}:${port}`,
-            credentials: this.s3sourceCredentials,
+        this._ringReader = null;
+        this._s3Client = null;
+        this._setupClients(sourceConfig);
+    }
+
+    /**
+     * Setup internal clients: `this._ringReader`, `this._s3Client`
+     * @param {object} sourceConfig - source config (also called bucketdConfig)
+     * @return {undefined}
+     */
+    _setupClients(sourceConfig) {
+        const { https, host, port } = sourceConfig;
+        const protocol = https ? 'https' : 'http';
+        const endpoint = `${protocol}://${host}:${port}`;
+        const s3sourceCredentials = new AWS.Credentials({
+            accessKeyId: sourceConfig.auth.accessKey,
+            secretAccessKey: sourceConfig.auth.secretKey,
+        });
+
+        this._ringReader = new BackbeatClient({
+            endpoint,
+            credentials: s3sourceCredentials,
             sslEnabled: protocol === 'https',
-            httpOptions: { agent: this.sourceHTTPAgent, timeout: 0 },
+            httpOptions: {
+                agent: new http.Agent({ keepAlive: true }),
```

Comment:
You have to setup https.Agent if protocol === 'https'

Reply:
I think I need your input on implementing this. For https, do we require the remote source certificates?